### PR TITLE
mark @react-native-async-storage/async-storage as optional dependency

### PR DESCRIPTION
**Pull request by neerfri**
## TL;DR

This PR marks `@react-native-async-storage/async-storage` as optional dependency.

## Why

When following the [setup instructions for solana's wallet adapter for dApps](https://github.com/solana-labs/wallet-adapter/blob/master/APP.md) it suggests to run the following command in order to install all the required packages:

```
npm install --save \
    @solana/wallet-adapter-base \
    @solana/wallet-adapter-react \
    @solana/wallet-adapter-react-ui \
    @solana/wallet-adapter-wallets \
    @solana/web3.js \
    react
```

The problem is that npm after version 7 will always install all peer dependencies which causes `react-native` to be installed and with it roughly 900 more package (🤯 ...). The only workaround right now is to use `--use-legacy-peer-deps` which is really not recommended and will hit webapp project maintainers.

Since users that use this in a react-native app will already have react-native in their `package.json` this should not be a problem and the version restriction will still be in effect.

You can check this by creating a new folder, running `npm init` (with a recent npm version) and the above `npm install ...` command. You will see `react-native` being installed with many more packages. You can then use `npm explain react-native` to see why it is installed. You can also try the install with `--use-legacy-peer-deps` to see the difference.

### CODE REVIEW

**Comment by Funkatronics** [comment]
HI @neerfri, hanks for the PR. this seems a reasonable change, I am just doing a few test builds to verify how things work with this change. i'll approve shortly if everything goes well

**Review by Funkatronics — APPROVED** [review]

**Comment by neerfri** [comment]
Thank you @sdlaver and @Funkatronics
